### PR TITLE
render pie chart with echarts option object

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Part of https://github.com/metabase/metabase/issues/33281

### Description

Creates the echarts `option` object and renders the pie chart using it.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Create a question
2. Set visualization type to `Pie 2`
3. Add to a dashboard
4. Send in an email subscription

### Demo

Output from subscription email (changed default colors with setting).

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

_Nothing to test at the moment. We will update percy visual tests later to test the look of the chart._

### CODE REVIEW

**Comment by EmmadUsmani** [comment]
Current dependencies on/for this PR:
* master
  * **PR #34606** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34606" target="_blank">Graphite</a> 
    * **PR #34717** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34717" target="_blank">Graphite</a> 
      * **PR #34772** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34772" target="_blank">Graphite</a> 
        * **PR #34809** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34809" target="_blank">Graphite</a> 
          * **PR #34816** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34816" target="_blank">Graphite</a>  👈
    * **PR #34703** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34703" target="_blank">Graphite</a> 
      * **PR #34763** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34763" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/34816?utm_source=stack-comment).

File: frontend/src/metabase/visualizations/echarts/pie/option.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// static, non-computed values
+const SUNBURST_SERIES_OPTIONS: RegisteredSeriesOption["sunburst"] = {
+  type: "sunburst",
+  nodeClick: false,
+  sort: undefined,
+  emphasis: { focus: "none" },
+  label: {
+    rotate: 0,
+    overflow: "none",
+    lineHeight: 50, // TODO confirm
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/pie/option.ts:19]: Later when polishing will get the exact values from existing chart and confirm with designers

File: frontend/src/metabase/visualizations/echarts/pie/option.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  radius: ["60%", "90%"], // TODO compute this dynamically based on side length like in PieChart.jsx
+};
+
+export function getPieChartOption(
+  chartModel: PieChartModel,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+): EChartsOption {
+  return {
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/pie/option.ts:31]: I'm intentionally leaving the options object sparse in this PR. Later PRs will focus on adding specific parts of the object, e.g. show total setting, show percentages setting, etc

File: frontend/src/metabase/visualizations/echarts/pie/option.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import type { PieChartModel } from "./model/types";
+
+// static, non-computed values
+const SUNBURST_SERIES_OPTIONS: RegisteredSeriesOption["sunburst"] = {
+  type: "sunburst",
+  sort: undefined,
+  label: {
+    rotate: 0,
+    overflow: "none",
+    lineHeight: 50, // TODO update this later
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/pie/option.ts:17]: Will get the exact values later when polishing, and will confirm it looks good with product+design

File: frontend/src/metabase/visualizations/echarts/pie/option/index.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  chartModel: PieChartModel,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+): EChartsOption {
+  return {
+    textStyle: {
+      fontFamily: renderingContext.fontFamily,
+    },
+    series: {
+      ...SUNBURST_SERIES_OPTIONS,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/pie/option/index.ts:21]: I'm intentionally leaving the options object sparse in this PR, my intention here is just setting the foundation. Later PRs will focus on adding specific parts of the object, e.g. show total setting, show percentages setting, etc

File: frontend/src/metabase/visualizations/echarts/pie/option/constants.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import type { RegisteredSeriesOption } from "echarts";
+
+export const SUNBURST_SERIES_OPTIONS: RegisteredSeriesOption["sunburst"] = {
+  type: "sunburst",
+  sort: undefined,
+  label: {
+    rotate: 0,
+    overflow: "none",
+    lineHeight: 50, // TODO update this later
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/pie/option/constants.ts:9]: Will get the exact values later when polishing, and will confirm it looks good with product+design

File: frontend/src/metabase/visualizations/echarts/pie/option/index.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      data: chartModel.slices.map(s => ({
+        value: s.value,
+        name: s.key,
```
Comment (alxnddr) [inline frontend/src/metabase/visualizations/echarts/pie/option/index.ts:24]: I think it should be a slice percentage here when the related settings enabled. For now we don't show dimension values in slices

**Review by alxnddr — APPROVED** [review]
One minor comment, can be addressed in further PRs

File: frontend/src/metabase/visualizations/echarts/pie/option/index.ts (hunk 6 above)
Comment (EmmadUsmani) [reply frontend/src/metabase/visualizations/echarts/pie/option/index.ts:24]: Yup I fix this in https://github.com/metabase/metabase/pull/34869

**Pull request closed without merging**